src/main/java/ru/vidtu/ias/crypt/HardwareCrypt.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * In-Game Account Switcher is a mod for Minecraft that allows you to change your logged in account in-game, without restarting Minecraft.
+ * Copyright (C) 2015-2022 The_Fireplace
+ * Copyright (C) 2021-2024 VidTu
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>
+ */
+
+package ru.vidtu.ias.crypt;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Crypt with "hardware ID".
+ *
+ * @author VidTu
+ */
+public final class HardwareCrypt implements Crypt {
+    /**
+     * Shared "hardware ID" crypt.
+     *
+     * @apiNote Use {@link #equals(Object)} for comparison
+     */
+    public static final HardwareCrypt INSTANCE = new HardwareCrypt();
+
+    /**
+     * Logger for this class.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger("IAS/HardwareCrypt");
+
+    /**
+     * Empty byte array for unknown MAC.
+     */
+    private static final byte[] EMPTY_MAC = {};
+
+    /**
+     * List of environmental keys used for hardware password.
+     */
+    private static final List<String> ENV = List.of("COMPUTERNAME", "PROCESSOR_ARCHITECTURE",
+            "PROCESSOR_REVISION", "PROCESSOR_IDENTIFIER", "PROCESSOR_LEVEL", "NUMBER_OF_PROCESSORS", "OS", "USERNAME",
```

Comment:
Has "OS" been checked on various operating systems to verify that it's typically a high level description of the OS and not going to change with, say, each minor update?

Reply:
On Windows (both 10 and 11) this returns consistent "WINDOWS_NT" string.
On Linux (from my testing of distros) it's either absent, returns "Linux", or returns the distro name like "Fedora", without versions of kernel or distro.
I don't own any Apple devices. according to the stuff on the internet, the OS env is not set on macOS.